app/assets/javascript/poke.js

```diff
@@ -44,7 +44,9 @@ Poke.prototype.load = function(poke) {
     	/* Change mega forme to base + item, and so on */
     	var item = PokeInfo.itemForForme(this);
     	if (item) {
-    		this.forme = 0;
+	        if (ItemInfo.isMegaStone(item)) {
+				this.forme = 0;
```

Comment:
Otherwise formes will default back to the base forme, e.g. you select Arceus-Dark, but Arceus normal is shown in teambuilder and this might confuse many users.